forum/qa-content/qa-protect-submit.js

```diff
@@ -0,0 +1,14 @@
+// Protect from multiple form submit
+
+window.addEventListener('DOMContentLoaded', function()
+{
+	$("#__form").submit(function()
+	{
+		$(this).submit(function()
```

Comment:
Ciekawe czy tego nie można jeszcze jakoś skrócić, nie bardzo ogarniam po co najpierw submit formularza, a jeszcze później submit na this.
Może ktoś ogarniający lepiej JS z @CodersCommunity/users by podpowiedział.

Reply:
Najwyraźniej nie rozumiesz zasady działania tego kodu. Przy pierwszym kliknięciu na przycisk wywołuje się zewnętrzna funkcja submit z jQuery. Ona zwraca true, czyli że request ma być wykonany, ale także nadpisuje samą siebie, aby zwracała false. Wynika z tego, że kolejne kliknięcia już nie wysyłają requesta.

Reply:
No nie do końca. Pierwsze kliknięcie wykona funkcję, która zwróci `true` i **doda** drugi listener, który będzie zwracał `false`. Drugie kliknięcie wykona funkcję, która zwróci `true` i **doda** trzeci listener, który będzie zwracał `false`, a następnie wykona drugi listener, który zwróci `false` itd.

Efekt, jaki chcemy osiągnąć, da nam albo `$.fn.once`, albo po prostu nadanie `[disabled]` przycisku wysyłania formularza.

Reply:
Ja wziąłem to rozwiązanie z internetu i sądziłem, że jest poprawne. Jeśli dopisanie disabled w JS jest prostsze, to można zastosować ten pomysł. Dzięki @Comandeer :)